scripts/mysql2-select-healthcheck

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+# NOTE: The zabbix user requires SELECT privileges on all databases/fields in
+# order for this to work.
+
+if test -s "/etc/zabbix/config/mysql2.$1.cnf"; then
+    cnf=/etc/zabbix/config/mysql2.$1.cnf
+elif test -s "/etc/zabbix/config/mysql.$1.cnf"; then
+    cnf=/etc/zabbix/config/mysql.$1.cnf
+elif test "$1" = default && test -s /etc/zabbix/mysql/.my.cnf; then
+    cnf=/etc/zabbix/mysql/.my.cnf
+else
+    echo "ZBX_ERROR: missing instance $1 cnf" >&2
+    exit 1
+fi
+
+system_tables="information_schema performance_schema sys"
+exclude_dbs=$(awk '
+    /^\[zabbix\]/ { in_section=1; next }
+    /^\[/ { in_section=0 }
+    in_section && /^exclude_healthcheck_dbs=/ {
+        split($0, a, "="); print a[2]; exit
+    }
+    ' $cnf)
+
+# If we want to check system tables, we'll need PROCESS privileges
+# ERROR 1227 (42000) at line 1: Access denied; you need (at least one of) the
+# PROCESS privilege(s) for this operation
+include_system_tables=false
+if test -n "$2" && test "$2" = 'include_system_tables'; then
+    include_system_tables=true
+fi
+
+for db in $(mysql --defaults-file="$cnf" -NB -e 'show databases;'); do
+    # ignore if the current db is a system table
+    if test "$include_system_tables" = "false"; then
+        if echo "$system_tables" | grep -wq "$db"; then
+            continue
+        fi
+    fi
+
+    if test -n "$exclude_dbs"; then
+        if echo "$exclude_dbs" | grep -wq "$db"; then
+            continue
+        fi
+    fi
+
+    for table in $(mysql --defaults-file="$cnf" "$db" -NB -e 'SHOW TABLES;'); do
+        table_type=$(mysql --defaults-file="$cnf" -N -B -e \
+            "SELECT TABLE_TYPE FROM information_schema.tables WHERE table_schema = '$db' AND table_name = '$table';")
+        if test "$table_type" = "VIEW"; then
+            continue
+        fi
```

Comment:
Maybe add `table_type = 'BASE TABLE'` to query and skip this?

So instead of SHOW TABLES, you'll do:
```
SELECT CONCAT(table_schema, '.', table_name) AS tbl FROM information_schema.tables WHERE table_type = 'BASE TABLE';
```
Where you get all DBs and all tables in a single query.

If you do this, you may want to refactor the DB usage/selection/exclusion as well.